tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenCreateIndexTest.cs

```diff
@@ -44,40 +47,48 @@ public override void Arrange(BsonDocument document)
             base.Arrange(document);
         }
 
+        // protected methods
+        protected override void AssertResult()
+        {
+        }
+
         protected override void CallMethod(CancellationToken cancellationToken)
         {
             if (_session == null)
             {
-                _collection.Indexes.CreateOne(_createIndexModel, new CreateOneIndexOptions(), cancellationToken);
+                _collection.Indexes.CreateOne(CreateIndexModel, new CreateOneIndexOptions(), cancellationToken);
             }
             else
             {
-                _collection.Indexes.CreateOne(_session, _createIndexModel, new CreateOneIndexOptions(), cancellationToken);
+                _collection.Indexes.CreateOne(_session, CreateIndexModel, new CreateOneIndexOptions(), cancellationToken);
             }
         }
 
         protected override async Task CallMethodAsync(CancellationToken cancellationToken)
         {
             if (_session == null)
             {
-                await _collection.Indexes.CreateOneAsync(_createIndexModel, new CreateOneIndexOptions(), cancellationToken);
+                await _collection.Indexes.CreateOneAsync(CreateIndexModel, new CreateOneIndexOptions(), cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                await _collection.Indexes.CreateOneAsync(_session, _createIndexModel, new CreateOneIndexOptions(), cancellationToken);
+                await _collection.Indexes.CreateOneAsync(_session, CreateIndexModel, new CreateOneIndexOptions(), cancellationToken).ConfigureAwait(false);
             }
         }
 
         protected override void SetArgument(string name, BsonValue value)
         {
             switch (name)
             {
+                case "keys":
+                    _indexKeys = value.AsBsonDocument;
+                    return;
+                case "name":
+                    _indexName = value.AsString;
+                    return;
```

Comment:
Refactor SetArgument to:

    protected override void SetArgument(string name, BsonValue value)
    {
        switch (name)
        {
            case "keys":
                _keys = value.AsBsonDocument;
                return;
            case "name":
                _name = value.AsString;
                return;
            case "session":
                _session = (IClientSessionHandle)_objectMap[value.AsString];
                return;
        }

        base.SetArgument(name, value);
    }

Reply:
Done.